spec/streamflow/pm.md

```diff
@@ -0,0 +1,416 @@
+
+| Field                       | Type                         | Description                                                                               |
+| --------------------------- | ---------------------------- | ----------------------------------------------------------------------------------------- |
+| **fundsAtCreation**         | uint256                      | Funds that the reserve was created with                                                   |
+| **fundsRemaining**          | uint256                      | Funds remaining in the reserve taking into account funds claimed if the reserve is frozen |
+| **freezeRound**             | uint256                      | The round that the reserve is frozen because the sender overspent from its deposit        |
+| **recipientsInFreezeRound** | uint256                      | Number of recipients (i.e. registered orchestrators) during `freezeRound`                 |
+| **claimed**                 | mapping (address => uint256) | Amount claimed by an ETH address (only eligible recipients during `freezeRound`)          |
+
+A broadcaster only has a single reserve at any given point in time.
+
+A broadcaster's reserve can be thought of as a fixed amount of funds that is committed to the set of orchestrators which is updated each round. So, at the beginning of each round, the reserve is split into equal allocations based on the current orchestrator set. Thus, without adding additional funds to a reserve, it will be split into smaller allocations as the orchestrator set approaches its maximum size and it will be split into larger allocations as the orchestrator set shrinks in size. Since each allocation represents an orchestrator's max float for the broadcaster, as the allocation size increase, an orchestrator will be able to safely receive winning tickets with higher face values or more winning tickets with lower face values prior to having to redeem them.
+
+An orchestrator can accept or reject work from a broadcaster based upon the broadcaster's reserve and the value of the allocation from the reserve committed to the orchestrator. For example, since the value of the allocation affects the maximum face value that can be used for tickets, lower reserves would require lower ticket face values and higher winning probabilities - if the winning probability is too high such that an orchestrator would need to redeem winning tickets too frequently (potentially contributing to network congestion), then an orchestrator might reject work from the broadaster.
```

Comment:
Late to the party here, but some comments:

> For example, since the value of the allocation affects the maximum face value that can be used for tickets, lower reserves would require lower ticket face values and higher winning probabilities

This part seems implementation specific based on whatever strategy O picks. One simpler strategy would be for an O to have a fixed faceValue / winProb for all streams. If `faceValue < float` then O can either decline to work with the B, or adjust faceValue / winProb to match the float.
 
O isn't necessarily concerned with a minimum number of winning tickets, only that it has enough to cover the faceValue of the current tickets being submitted.

Stream continuity is a concern for B, but not for an O. The onus is on B to ensure that its stream can continue to be transcoded uninterrupted. O just wants to get paid given the current ticket parameters.

B could apply an internal heuristic during O selection to ensure that its float is sufficient to cover a certain number of tickets, but this doesn't need to be something that the O specifies. However, this might not be necessary given current failover behavior of the node. O would decline to  do further work for B if the outstanding winning tickets had the potential to drain B's float. B would just fail over to the next O, and transcoding would continue uninterrupted.

> too high such that an orchestrator would need to redeem winning tickets too frequently (potentially contributing to network congestion), then an orchestrator might reject work from the broadaster.

As mentioned [elsewhere](https://github.com/livepeer/research/issues/29#issuecomment-479688140), gas overhead is more of a concern for B - they do not want to subsidize too many transactions on the network. On the other hand, O can incorporate the gas overhead into the faceValue, so they are more decoupled from the effects of network congestion on gas pricing. The O is assured a certain amount of revenue regardless of gas overhead.

Reply:
> This part seems implementation specific based on whatever strategy O picks. One simpler strategy would be for an O to have a fixed faceValue / winProb for all streams. If faceValue < float then O can either decline to work with the B, or adjust faceValue / winProb to match the float.

Good point! I can update the spec to clarify that O's strategy for determining whether to work with a B will depend on the implementation.

> If faceValue < float then O can either decline to work with the B

Do you mean if `faceValue > float`?

> Stream continuity is a concern for B, but not for an O. The onus is on B to ensure that its stream can continue to be transcoded uninterrupted. O just wants to get paid given the current ticket parameters.

This is true, but I also see a benefit in having the default O strategy guaranteeing that O can receive at least 1 winning ticket without interrupting B's stream in order to ensure that a B won't switch away from O just because B sent a winning ticket to O with a face value that matches the `maxFloat` for B.

> On the other hand, O can incorporate the gas overhead into the faceValue, so they are more decoupled from the effects of network congestion on gas pricing. The O is assured a certain amount of revenue regardless of gas overhead.

This is true, but the cited quote focuses on the frequency of winning ticket redemptions as opposed to the profitability of winning ticket redemptions. While O could certainly be happy in the short term as long as winning ticket redemptions are profitable (due to the transaction cost being priced into the ticket's face value), in the medium/long term frequent winning ticket redemptions by O can contribute to network congestion which is a negative externality that affects all ETH users that also affects O because the higher transaction cost (as a result of increased network congestion) could (depending on O's strategy) push the required ticket face value higher and as a result the number of Bs that O can accept tickets from might decrease. I think there is a benefit in having O's default strategy try to avoid very frequent winning ticket redemptions when defining required ticket parameters.